cdap-app-fabric/src/test/java/co/cask/cdap/gateway/handlers/InMemoryAuthorizer.java

```diff
@@ -46,22 +61,96 @@ public void enforce(EntityId entity, Principal principal, Action action) throws
 
   @Override
   public void grant(EntityId entity, Principal principal, Set<Action> actions) {
-    get(entity, principal).addAll(actions);
+    getActions(entity, principal).addAll(actions);
   }
 
   @Override
   public void revoke(EntityId entity, Principal principal, Set<Action> actions) {
-    get(entity, principal).removeAll(actions);
+    getActions(entity, principal).removeAll(actions);
   }
 
   @Override
   public void revoke(EntityId entity) {
     for (Principal principal : table.row(entity).keySet()) {
-      get(entity, principal).clear();
+      getActions(entity, principal).clear();
+    }
+  }
+
+  @Override
+  public void createRole(Role role) throws RoleAlreadyExistsException {
+    if (roleToPrincipalMap.containsKey(role)) {
+      throw new RoleAlreadyExistsException(role);
+    }
+    roleToPrincipalMap.put(role, new HashSet<Principal>());
+  }
+
+  @Override
+  public void dropRole(Role role) throws RoleNotFoundException {
```

Comment:
We can but let's think about it. Why should I not be able to drop a role which has users or principals in it ? 
Let's say I have a role PurchaeDSReaders I want to drop this role so I should first go and remove everyone from it and then drop it ? Seems like unnecessary work. I think a user who is deleting a role is wants to delete it so that no one can for that privilege anymore.
@anew : Do you have any suggestions ?